### PR TITLE
Allow the user to implement custom HTML

### PR DESCRIPTION
**Pull request by alexanderturinske**
- Inspired by #24
- The HTML MUST have certain classes to not break the library
  - title
  - description
  - copyright
  - versions

### CODE REVIEW

**Comment by rhysd** [comment]
Let me confirm one thing before review. `use_inner_html` does not work for your usecase?

**Comment by bampakoa** [comment]
@rhysd How is `use_inner_html` related with the **versions** part?

**Comment by alexanderturinske** [comment]
@rhysd The idea of this is that the user can rearrange the HTML in any way they want, making the library more malleable. `use_inner_html` only applies to copyright, license, and description, while this option would allow for customizability with the bug link, icon, product name, etcetera.

File: README.md [hunk 1]
```diff
@@ -37,6 +38,7 @@ export default function openAboutWindow(info: {
     css_path?: string | string[];
     adjust_window_size?: boolean;
     win_options?: BrowserWindowOptions;
+    html_path?: string;
```
Comment (rhysd) [inline README.md]: Could you rename this option to more specific name `custom_html_path`?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "about-window",
-  "version": "1.11.0",
+  "version": "1.12.0",
```
Comment (rhysd) [inline package.json]: Please remove this line since I bump up version when publishing new version.

File: README.md [hunk 3]
```diff
@@ -86,6 +88,7 @@ $ npm run debug
 | `use_inner_html` | If `true`, set the value with `.innerHTML` on copyright, license and description Default is `false`. **Optional** | boolean |
 | `bug_link_text` | Text for a bug report link. **Optional** | string |
 | `product_name` | Name of the application **Optional** | string |
+| `html_path` | Path to user-defined HTML file. It needs to include the following classes: title, description, copyright, and versions. **Optional** | string |
```
Comment (rhysd) [inline README.md]: > It needs to include the following classes: title, description, copyright, and versions.

I think we can avoid this restriction as follows:

```diff
diff --git a/src/renderer.ts b/src/renderer.ts
index e731697..38deb17 100644
--- a/src/renderer.ts
+++ b/src/renderer.ts
@@ -7,38 +7,50 @@ ipcRenderer.on('about-window:info', (_: any, info: AboutWindowInfo) => {
     document.title = `About ${app_name}`;
 
     const title_elem = document.querySelector('.title') as HTMLHeadingElement;
-    title_elem.innerText = `${app_name} ${remote.app.getVersion()}`;
+    if (title_elem !== null) {
+        title_elem.innerText = `${app_name} ${remote.app.getVersion()}`;
+    }
 
     if (info.homepage) {
         title_elem.addEventListener('click', open_home);
         title_elem.classList.add('clickable');
         const logo_elem = document.querySelector('.logo');
-        logo_elem.addEventListener('click', open_home);
-        logo_elem.classList.add('clickable');
+        if (logo_elem !== null) {
+            logo_elem.addEventListener('click', open_home);
+            logo_elem.classList.add('clickable');
+        }
     }
 
     const copyright_elem = document.querySelector('.copyright') as any;
-    if (info.copyright) {
-        copyright_elem[content] = info.copyright;
-    } else if (info.license) {
-        copyright_elem[content] = `Distributed under ${info.license} license.`;
+    if (copyright_elem !== null) {
+        if (info.copyright) {
+            copyright_elem[content] = info.copyright;
+        } else if (info.license) {
+            copyright_elem[content] = `Distributed under ${info.license} license.`;
+        }
     }
 
     const icon_elem = document.getElementById('app-icon') as HTMLImageElement;
-    icon_elem.src = info.icon_path;
+    if (icon_elem !== null) {
+        icon_elem.src = info.icon_path;
+    }
 
     if (info.description) {
         const desc_elem = document.querySelector('.description') as any;
-        desc_elem[content] = info.description;
+        if (desc_elem !== null) {
+            desc_elem[content] = info.description;
+        }
     }
 
     if (info.bug_report_url) {
         const bug_report = document.querySelector('.bug-report-link') as HTMLDivElement;
-        bug_report.innerText = info.bug_link_text || 'Report an issue';
-        bug_report.addEventListener('click', e => {
-            e.preventDefault();
-            shell.openExternal(info.bug_report_url);
-        });
+        if (bug_report !== null) {
+            bug_report.innerText = info.bug_link_text || 'Report an issue';
+            bug_report.addEventListener('click', e => {
+                e.preventDefault();
+                shell.openExternal(info.bug_report_url);
+            });
+        }
     }
 
     if (info.css_path) {
```

**Comment by rhysd** [comment]
Thank you for the answer. Let me review this patch.

**Comment by rhysd** [comment]
Closing this as inactive

**Pull request closed without merging**